xds/src/main/java/io/grpc/xds/EdsLoadBalancer2.java

```diff
@@ -108,6 +110,10 @@ public void handleResolvedAddresses(ResolvedAddresses resolvedAddresses) {
       xdsClientPool = resolvedAddresses.getAttributes().get(XdsAttributes.XDS_CLIENT_POOL);
       xdsClient = xdsClientPool.getObject();
     }
+    if (callCounterProvider == null) {
+      callCounterProvider =
+          resolvedAddresses.getAttributes().get(XdsAttributes.CALL_COUNTER_PROVIDER);
```

Comment:
If `childLbState` does not need need it, might discard the key for downstream balancers.

Reply:
Well, not necessary. Just like the object pool for XdsClient, we do not discard it even if it is not needed in the leaf LB policy.